api/src/lib.rs

```diff
@@ -593,18 +607,20 @@ impl Client {
         request: RequestT,
     ) -> Result<SuccessT>
     where
-        LocationT: IntoUrl + Display,
+        LocationT: IntoUrl + Display + Clone,
         RequestT: Serialize,
         for<'de> SuccessT: Deserialize<'de>,
         for<'de> ErrorT: Deserialize<'de> + ApiError,
     {
         debug!("Attempting POST `{}`", url);
         let http_response = self
-            .http_client
-            .post(url)
-            .headers(self.headers.clone())
-            .json(&request)
-            .send()
+            .handle_timeouts(|| {
+                self.http_client
+                    .post(url.clone())
+                    .headers(self.headers.clone())
+                    .json(&request)
+                    .send()
+            })
```

Comment:
Our requests are generally idempotent, but a blindly retrying a `POST` request seems a bit dangerous---should we make this a parameter to `post` so that each route can decide for itself whether it's safe to retry?